src/group-node.cpp

```diff
@@ -173,6 +173,11 @@ namespace graphics {
         }
     }
 
+    osg::Node* GroupNode::getOsgNode() const
+    {
+      return (*(list_of_objects_.begin()))->getOsgNode();
```

Comment:
I would rather return the last OSG node of the group, by using:
```cpp
return this->asQueue();
```
because
```cpp
(*(list_of_objects_.begin()))->getOsgNode();
```
belongs to a child `graphics::Node`, not to this `GroupNode`.

With your implementation, it is not possible to attach a camera to an empty `GroupNode`, which is a bug.